packages/kbn-test/src/jest/setup/react_testing_library.js

```diff
@@ -19,3 +19,15 @@ import { configure } from '@testing-library/react';
 
 // instead of default 'data-testid', use kibana's 'data-test-subj'
 configure({ testIdAttribute: 'data-test-subj', asyncUtilTimeout: 4500 });
+
+/* eslint-env jest */
+
+jest.mock('@testing-library/react', () => {
+  const actual = jest.requireActual('@testing-library/react');
+
+  return {
+    ...actual,
+    render: (ui, options) => actual.render(ui, { ...options, legacyRoot: true }),
```

Comment:
Forcing `legacyRoot:true` by default makes testing library to use legacy API ReactDOM.render api which is how we make it work with React@17. 
The option is documented here https://testing-library.com/docs/react-testing-library/api/#legacyroot . 

There is no global configuration for this option, so we have to do this mocking workaround to adjust the default. 

Event after upgrade of the react packages to React@18 it would make sense to keep this option as it until we start upgrading to the Concurrent Mode. We will likely come of a gradual transition plan in parallel with the runtime.

Reply:
Yeah same thing here as above, I feel at minimum we need a comment to track this.